### PR TITLE
Update Radio component

### LINKED ISSUE #2710: Update the Radio button component
**AC:**
- [ ] Implement all the updates
- [ ] Create the `RadioButtonGroup` component that can manage state of multiple radio buttons and return a single array of values. `<RadioButtonGroup value={categoriesFilter} options={[]} onChange={(values) => void} />`

---

The goal of this task is to update the Radio button component based on the designs, docs, and changelog, which [can be found in Figma](https://www.figma.com/file/Pf31tuYpozYmpq163U2ho8/Web-components?node-id=7384%3A142876). 

>  **Changelog (from Figma)**
> - Updated fill, border and text colors to use color design tokens,
> - Updated transition animations to use animation design tokens,
> - Rearranged and renamed component properties,
> - Added Styling property with Error variants (for selected States),
> - Added “Radio button group” component for easier and more consistent vertical stacking,
> - Added documentation along with content/usage guidelines,
> - Added Focus state, which inherits its styling from the Hover state.

image

### PR DESCRIPTION
**Pull request by rafalpawlow**
Closes #2710 

**Design**
https://www.figma.com/file/Pf31tuYpozYmpq163U2ho8/Web-components?node-id=7384%3A142876

### CODE REVIEW

**Review by toiletgranny — CHANGES_REQUESTED** [review]
Perfect, same comments as with #2789: 
> 1. The helper text is still `var(--color-core-neutral-300)`, should be `var(--color-text)`,
> 2. This changelog item seems to not be implemented: "Added Focus state, which inherits its styling from the Hover state".  The goal here is to do the same thing that @drillprop has done with the Switch component recently. Ideally, we use the the `:focus-visible`, instead of `:focus`, if that's okay with you.

**Review by drillprop — CHANGES_REQUESTED** [review]
The `:focus-visible` issue is still not fixed. Use`@supports` media query to make this work. 
Basically, if the browser has support for `:focus-visible` use `:focus-visible`. if not use regular `:focus`.  

Check my code in the Switch component. Btw, maybe it's possible to write this cleaner and more readable, but I couldn't figure out the better way.

**Comment by drillprop** [comment]
Crap, I just noticed that you used `:focus-within`. Unfortunately, it's not possible to use `:focus-within` together with `:focus-visible` - https://github.com/WICG/focus-visible/issues/151. You'll need to rebuild HTML a bit.

**Comment by drillprop** [comment]
I just made a small experiment on how it would work with [this package](https://www.npmjs.com/package/what-input). Check the code here https://github.com/drillprop/atlas/commit/993d5ca244718a11f9ad8154eae90b0967661acc

**Comment by toiletgranny** [comment]
I'm afraid we lost the option to select a radio button by clicking on its label.

**Review by toiletgranny — APPROVED** [review]

File: packages/atlas/src/components/_inputs/RadioButtonGroup/RadioButtonGroup.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+export type RadioButtonGroupProps = {
+  options: RadioButtonProps[]
+  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void
+  value: string | number
+  name?: string
+  disabled?: boolean
+  error?: boolean
+}
+
+export const RadioButtonGroup: React.FC<RadioButtonGroupProps> = ({
```
Comment (drillprop) [inline packages/atlas/src/components/_inputs/RadioButtonGroup/RadioButtonGroup.tsx]: Can we actually make use of this component and see it in action? We could apply this in the VideoForm at least?

File: packages/atlas/src/components/_inputs/RadioInput/RadioInput.styles.ts [hunk 2]
```diff
@@ -1,139 +1,163 @@
...
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  background-clip: content-box;
+  padding: 4px;
+  box-sizing: border-box;
+  border: 1px solid ${({ error }) => cVar(error ? 'colorBorderError' : 'colorBorderAlpha')};
+  transition: all ${cVar('animationTransitionFast')};
+
+  ::before {
```
Comment (drillprop) [inline packages/atlas/src/components/_inputs/RadioInput/RadioInput.styles.ts:27]: Could you add a comment in the code that describes what `:before` and `:after` is? Just add something like that:
```tsx
/* inner dot */ 
:after
```

File: packages/atlas/src/components/_inputs/RadioInput/RadioInput.styles.ts [hunk 3]
```diff
@@ -1,139 +1,163 @@
...
-  checked?: boolean
-  error?: boolean
-}
+  position: relative;
+  border-radius: 50%;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  background-clip: content-box;
+  padding: 4px;
```
Comment (drillprop) [inline packages/atlas/src/components/_inputs/RadioInput/RadioInput.styles.ts]: let's use sizes here

File: packages/atlas/src/components/_inputs/RadioInput/RadioInput.styles.ts [hunk 4]
```diff
@@ -1,139 +1,163 @@
...
 
-    &::before {
-      background-color: ${cVar('colorCoreNeutral800Lighten')};
+  position: absolute;
+  top: 0;
+  left: 0;
+  margin: auto;
+  opacity: 0;
+  cursor: ${(props) => (props.disabled ? 'not-allowed' : 'pointer')};
+  z-index: 999;
```
Comment (drillprop) [inline packages/atlas/src/components/_inputs/RadioInput/RadioInput.styles.ts]: We could use `zIndex.globalOverlay` here, but on the other hand, Is this value too big?

**Review by drillprop — CHANGES_REQUESTED** [review]

File: packages/atlas/src/components/_inputs/RadioButtonGroup/RadioButtonGroup.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    className: { table: { disable: true } },
+    selectedValue: { table: { disable: true } },
+    options: { table: { disable: true } },
+    onChange: { table: { disable: true } },
+    value: { table: { disable: true } },
+    name: { table: { disable: true } },
+  },
+  args: {
+    label: 'Hello there',
+    helperText: 't100',
```
Comment (drillprop) [inline packages/atlas/src/components/_inputs/RadioButtonGroup/RadioButtonGroup.stories.tsx]: https://github.com/Joystream/atlas/pull/2789#discussion_r893545149

File: packages/atlas/src/components/_inputs/RadioButtonGroup/RadioButtonGroup.tsx (hunk 1 above)
Comment (rafalpawlow) [reply packages/atlas/src/components/_inputs/RadioButtonGroup/RadioButtonGroup.tsx]: This was very good idea 👍

File: packages/atlas/src/components/_inputs/RadioButton/RadioButton.stories.tsx [hunk 6]
```diff
@@ -1,32 +1,27 @@
...
   title: 'inputs/RadioButton',
   component: RadioButton,
+  argTypes: {
+    className: { table: { disable: true } },
+    selectedValue: { table: { disable: true } },
+    selected: { controls: { type: 'boolean' } },
+  },
   args: {
     label: 'Hello there',
     helperText: 't100',
```
Comment (drillprop) [inline packages/atlas/src/components/_inputs/RadioButton/RadioButton.stories.tsx]: image
We still have `helperText` control which does nothing. I'm a little surprised why typescript doesn't mark that as an error.

File: packages/atlas/src/components/_inputs/RadioButton/RadioButton.tsx [hunk 7]
```diff
@@ -3,25 +3,19 @@ import React from 'react'
...
+  ({ value, selectedValue, label, caption, className, disabled, error, onChange, ...props }, ref) => {
     return (
-      <RadioAndCheckboxBase
-        disabled={disabled}
-        className={className}
-        label={label}
-        error={error}
-        helperText={helperText}
-      >
+      <RadioAndCheckboxBase disabled={disabled} className={className} label={label} error={error} helperText={caption}>
```
Comment (drillprop) [inline packages/atlas/src/components/_inputs/RadioButton/RadioButton.tsx]: Let's also rename helperText to caption here. Is there a reason why we keep that unchanged?

File: packages/atlas/src/views/studio/VideoWorkspace/VideoForm/VideoForm.tsx [hunk 8]
```diff
@@ -590,27 +589,18 @@ export const VideoForm: React.FC<VideoFormProps> = React.memo(({ onSubmit, setFo
...
-                    helperText={errors.isExplicit ? 'Content rating must be selected' : ''}
-                    disabled={videoFieldsLocked}
-                  />
-                </RadioButtonsContainer>
+                <RadioButtonGroup
+                  ref={ref}
+                  options={[
+                    { label: 'All audiences', value: 'false' },
+                    { label: 'Mature', value: 'true' },
+                  ]}
```
Comment (drillprop) [inline packages/atlas/src/views/studio/VideoWorkspace/VideoForm/VideoForm.tsx:678]: I set my video as mature, but nothing happened when I published my changes, and the transaction was completed. Could you check what's wrong?

**Review by drillprop — CHANGES_REQUESTED** [review]

File: packages/atlas/src/views/studio/VideoWorkspace/VideoForm/VideoForm.tsx (hunk 8 above)
Comment (drillprop) [reply packages/atlas/src/views/studio/VideoWorkspace/VideoForm/VideoForm.tsx:678]: ok, ignore that. It's not your fault. This should be fixed once we merge https://github.com/Joystream/atlas/pull/2854

**Comment by kdembler** [comment]
Needs to target `dev` (and optionally rebase)

**Review by drillprop — APPROVED** [review]

**Pull request merged**